cluster-api/pkg/apis/cluster/v1alpha1/machinedeployment_types.go

```diff
@@ -0,0 +1,288 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/endpoints/request"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1validation "k8s.io/apimachinery/pkg/apis/meta/v1/validation"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	utilvalidation "k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"k8s.io/kube-deploy/cluster-api/pkg/apis/cluster"
+	"k8s.io/kube-deploy/cluster-api/pkg/apis/cluster/common"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineDeployment
+// +k8s:openapi-gen=true
+// +resource:path=machinedeployments,strategy=MachineDeploymentValidationStrategy
+type MachineDeployment struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MachineDeploymentSpec   `json:"spec,omitempty"`
+	Status MachineDeploymentStatus `json:"status,omitempty"`
+}
+
+// MachineDeploymentSpec defines the desired state of MachineDeployment
+type MachineDeploymentSpec struct {
+	// Number of desired machines. Defaults to 1.
+	// This is a pointer to distinguish between explicit zero and not specified.
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// Label selector for machines. Existing MachineSets whose machines are
+	// selected by this will be the ones affected by this deployment.
+	// It must match the machine template's labels.
+	Selector metav1.LabelSelector `json:"selector"`
+
+	// Template describes the machines that will be created.
+	Template MachineTemplateSpec `json:"template"`
+
+	// The deployment strategy to use to replace existing machines with
+	// new ones.
+	// +optional
+	Strategy MachineDeploymentStrategy `json:"strategy,omitempty"`
+
+	// Minimum number of seconds for which a newly created machine should
+	// be ready.
+	// Defaults to 0 (machine will be considered available as soon as it
+	// is ready)
+	// +optional
+	MinReadySeconds *int32 `json:"minReadySeconds,omitempty"`
+
+	// The number of old MachineSets to retain to allow rollback.
+	// This is a pointer to distinguish between explicit zero and not specified.
+	// Defaults to 1.
+	// +optional
+	RevisionHistoryLimit *int32 `json:"revisionHistoryLimit,omitempty"`
+
+	// Indicates that the deployment is paused.
+	// +optional
+	Paused bool `json:"paused,omitempty"`
+
+	// The maximum time in seconds for a deployment to make progress before it
+	// is considered to be failed. The deployment controller will continue to
+	// process failed deployments and a condition with a ProgressDeadlineExceeded
+	// reason will be surfaced in the deployment status. Note that progress will
+	// not be estimated during the time a deployment is paused. Defaults to 600s.
+	ProgressDeadlineSeconds *int32 `json:"progressDeadlineSeconds,omitempty"`
+}
+
+// MachineDeploymentStrategy describes how to replace existing machines
+// with new ones.
+type MachineDeploymentStrategy struct {
+	// Type of deployment. Can be only be "RollingUpdate".
```

Comment:
Update comment "Can be only be" => "Can be only"?

Does it make sense to say that say that this is the only supported stragegy for now? I'd just not mention it since MachineDeploymentStrategyType already has the list.

Reply:
Done